import "trident/rpc_option.proto";

package trident.test;

option cc_generic_services = true;

message SleepRequest {
    required int32 sleep_time = 1; // in seconds
}

message SleepResponse {
    required string message = 1;
}

service SleepServer {
    // The service timeout is 2 seconds.
    option (trident.service_timeout) = 2000;

    rpc SleepWithServiceTimeout(SleepRequest) returns(SleepResponse);

    // The method timeout is 4 seconds.
    rpc SleepWithMethodTimeout(SleepRequest) returns(SleepResponse) {
        option (trident.method_timeout) = 4000;
    }
}
